.github/workflows/check.yml

```diff
@@ -71,3 +72,20 @@ jobs:
     - name: Build docs
       if: ${{matrix.build_docs}}
       run: cmake --build '${{github.workspace}}/build' --target doc
+
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v2
+      with:
+        name: build results ${{ matrix.os }} ${{ matrix.build }}
+        path: ${{github.workspace}}/build
+        if-no-files-found: error
+
+    - name: Upload rdiff
+      uses: actions/upload-artifact@v2
+      if: ${{ ! matrix.no_rdiff }}
+      with:
+        name: rdiff ${{ matrix.os }} ${{ matrix.build }}
```

Comment:
This is not enough to uniquely identify the particular job, as there is also an "options" dimension. This dimension means there are 4 different "os=ubuntu_latest,build=Release" runs with different options settings of "", "-DBUILD_SHARED_LIBS=OFF", "-DBUILD_RDIFF=OFF", and "-G Ninja -DCMAKE_C_COMPILER=clang".

I think this means it is currently building this artifact 4 times with each run and its a case of "last one is kept".

We probably want to keep all of them?

Reply:
I added the matrix.options dimension to the name.